lib/pymedphys/_streamlit/utilities/monaco.py

```diff
@@ -2,13 +2,114 @@
 # pylint: disable = no-value-for-parameter, expression-not-assigned
 # pylint: disable = too-many-lines, redefined-outer-name
 
+import os
 import pathlib
+from typing import Optional
 
 from pymedphys._imports import streamlit as st
 
+from pymedphys._monaco import patient as mnc_patient
+from pymedphys._streamlit.utilities import exceptions as _exceptions
+
 from . import exceptions, misc
 
 
+def monaco_tel_files_picker(
+    config: dict,
+    patient_id: str = "",
+    key_namespace: str = "",
+    advanced_mode: bool = False,
+    site: Optional[str] = None,
+    plan_selection_text: str = "",
+):
+    """A Streamit widget for selecting a Monaco plan tel file.
+
+    Parameters
+    ----------
+    config : dict
+        The user's configuration
+    patient_id : str, optional
+        A patient ID to default to, by default ""
+    key_namespace : str, optional
+        A string to prepend to all Streamlit widget keys, by default ""
+    advanced_mode : bool, optional
+        Whether or not to display information and options intended for
+        advanced users, by default False
+    site : str, optional
+        A site to default to, by default None
+    plan_selection_text : str, optional
+        Text to display to the user before the plan selection radio
+        button, by default ""
+
+    Returns
+    -------
+    monaco_site : pathlib.Path
+    monaco_directory : pathlib.Path,
+    patient_name : str,
+    selected_monaco_plan : str,
+    tel_paths : List[pathlib.Path],
+
+    """
```

Comment:
This function was refactored out from `monaco_input_method` for use within the electrons app.